From fd390ccb3bff46f9b25a6d94b3269fb7bd16c9a1 Mon Sep 17 00:00:00 2001
Message-Id: <fd390ccb3bff46f9b25a6d94b3269fb7bd16c9a1.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Tue, 18 Nov 2014 02:31:56 +0530
Subject: [PATCH 24/34] FOR_UPSTREAM [VPG] drm/i915: Update avi info frame for
 chv.

This patch updates the avi info frame for CHV. AVI info frame
communicates colorimetry, picture aspect ratio, pixel-repetition
factor, RGB or YCbCr indicator,and others specified by CEA standard.
This is required for HDMI compliance, I have tested this by using
astro analyzer.

Issue: GMINL-4129
Change-Id: Icf0d9646031537a2d0e62097fa75e3ddae16930e
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/intel_hdmi.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 39f967f..472613e 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -1671,9 +1671,12 @@ static void chv_hdmi_pre_enable(struct intel_encoder *encoder)
 {
 	struct intel_digital_port *dport = enc_to_dig_port(&encoder->base);
 	struct drm_device *dev = encoder->base.dev;
+	struct intel_hdmi *intel_hdmi = &dport->hdmi;
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct intel_crtc *intel_crtc =
 		to_intel_crtc(encoder->base.crtc);
+	struct drm_display_mode *adjusted_mode =
+		&intel_crtc->config.adjusted_mode;
 	enum dpio_channel ch = vlv_dport_to_channel(dport);
 	int pipe = intel_crtc->pipe;
 	int data, i;
@@ -1798,6 +1801,10 @@ static void chv_hdmi_pre_enable(struct intel_encoder *encoder)
 
 	mutex_unlock(&dev_priv->dpio_lock);
 
+	intel_hdmi->set_infoframes(&encoder->base,
+				   intel_crtc->config.has_hdmi_sink,
+				   adjusted_mode);
+
 	intel_enable_hdmi(encoder);
 
 	vlv_wait_port_ready(dev_priv, dport);
-- 
1.7.9.5

